mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -3800,17 +3800,37 @@ static bool haveSameTiles(PackOp packOp, UnPackOp unPackOp) {
   return true;
 }
 
-/// Fold an unpack(pack(x)) to x.
+/// Returns true if the pack op does not need a padding value.
+static bool paddingIsNotNeeded(PackOp op) {
+  auto srcType = op.getSourceType();
+  if (llvm::any_of(op.getInnerDimsPos(),
```

Comment:
What if the tiles are not statically known? It looks like we continue in `requirePaddingValue`; Can we also add a test for the tiles?

Reply:
This is why we check if those dim sizes are dynamic or not in the first beginning. I think we can only do nothing if they are dynamic tiles.

Reply:
oh you meant dynamic inner tile sizes? Good catch, adding the case now...